docs/admin/src/widgets/variationPreviewTemplate.js

```diff
@@ -12,31 +15,57 @@ const templateWithIcons = template.replace(
   ( match, icon ) => require( `../../../_includes/icons/${ icon }.svg` )
 );
 
+/**
+ * @param {MouseEvent} event - The mouse event object from the click.
+ */
+
+
 export default class Preview extends Component {
 
-  componentDidMount() {
+  constructor( props ) {
+    super( props );
     const entities = new AllHtmlEntities();
-    const tabs = document.querySelector( 'iframe' ).contentWindow.document.querySelectorAll( '[data-module="tabs"]' );
 
+    liquidEngine.registerFilter( 'slugify', initial => slugify( initial || '', { lower: true } ) );
     liquidEngine.registerFilter( 'xml_escape', initial => entities.encode( initial ) );
     liquidEngine.registerFilter( 'markdownify', initial => marked( initial || '' ) );
 
-    if ( tabs && tabs.length > 0 ) {
-      document.querySelector( 'iframe' ).contentWindow.document.body.classList.add( 'js-enabled' );
-      for ( let i = 0; i < tabs.length; i++ ) {
-        const tab = tabs[i];
-        new Tabs( tab ).init( );
-      }
+    this.containerRef = React.createRef();
+  }
+
+  handleClick( event ) {
+    const target = event.target;
+    // Returns the preview component's native DOM element for
+    // use with our non-React vanilla JS toggling and tabs functionality.
+    // https://reactjs.org/docs/react-dom.html#finddomnode
+    const doc = ReactDOM.findDOMNode( this );
```

Comment:
Do we need to look this up on every click, or are the DOM references known and can be stored?

Reply:
On second thought, I think we can get rid of it and just use the react container ref.